lib/linked_list.rb

```diff
@@ -12,120 +12,265 @@ def initialize(value, next_node = nil)
 
 # Defines the singly linked list
 class LinkedList
-    def initialize
-      @head = nil # keep the head private. Not accessible outside this class
+  def initialize
+    @head = nil # keep the head private. Not accessible outside this class
+  end
+  
+  # method to add a new node with the specific data value in the linked list
+  # insert the new node at the beginning of the linked list
+  def add_first(value)
+    @head = Node.new(value)

```

Comment:
You need to attach the new node to the rest of the list.

```suggestion
    @head = Node.new(value, @head)

```